GetStream/stream-chat-android
# [4226] Preview failed uploads

### 🎯 Goal

closes #4226

Failed video and image uploads have blank previews because they are trying to load from a `null` string instead of trying to upload from the local `Attachment.upload` file.

### 🛠 Implementation details

Implemented checks to see if the file has been uploaded or not and load the media accordingly.

### 🎨 UI Changes

| Before | After |
| --- | --- |
| ![Screenshot_1664812284](https://user-images.githubusercontent.com/37080097/193621891-daee8708-6922-4b7a-aab7-454fb82ad0cc.png) | ![Screenshot_1664812241](https://user-images.githubusercontent.com/37080097/193621928-c8eb86ab-a622-4856-9602-9e993452b45a.png) |

### 🧪 Testing

1. Launch the compose app
2. Go to any channel
3. Start uploading a video or an image and turn off the internet mid upload

Expected: You should see a failed upload and it should have a working preview

### ☑️Contributor Checklist

#### General

#### Code & documentation
- [ ] ~~Changelog is updated with client-facing changes~~ (fixing unreleased stuff)

### ☑️Reviewer Checklist

### 🎉 GIF

![gif](https://media3.giphy.com/media/12ETcAyWE6TjHy/giphy.gif?cid=ecf05e477z91cu9rdiq0f0s2aa76ma1gtovl2ea4wec1wo88&rid=giphy.gif&ct=g)
